File: Greedy/GasStation/gas.cpp
#include <vector>
using namespace std;

class Solution {
  public:
    int startStation(vector<int> &gas, vector<int> &cost) {
        //  code here
        int total = 0;
        int tank = 0;
        int start = 0;
        int n = gas.size();
        for(int i=0;i<n;i++){
            total += (gas[i]-cost[i]);
            tank += (gas[i] - cost[i]);
            if(tank<0){
                tank = 0;
                start = i+1;
            }
        }
        if(total<0)return -1;
        return start;
    }
};

File: Greedy/GasStation/gas.py
class Solution:
    def startStation(self, gas: list[int], cost: list[int]) -> int:
        total = 0
        tank = 0
        start = 0
        n = len(gas)

        for i in range(n):
            total += gas[i] - cost[i]
            tank += gas[i] - cost[i]

            if tank < 0:
                tank = 0
                start = i + 1

        if total < 0:
            return -1

        return start